File: Yufung-admin-front/src/services/activityLog.ts
import { request } from '@umijs/max';

export interface ActivityLog {
  id?: string;
  log_id?: string;
  user_id: string;
  username: string;
  company_id: string;
  company_name: string;
  operation_type: string;
  module_name: string;
  operation_desc: string;
  request_url: string;
  request_method: string;
  request_params: any;
  ip_address: string;
  user_agent: string;
  operation_time: string;
  execution_time: number;
  result_status: string;
  target_id?: string;
  target_name?: string;
}

export interface ActivityLogQuery {
  page?: number;
  page_size?: number;
  company_id?: string;
  user_id?: string;
  operation_type?: string;
  module_name?: string;
  result_status?: string;
  start_time?: string;
  end_time?: string;
  sort_by?: string;
  sort_order?: string;
}

export interface ActivityLogResponse {
  code: number;
  message: string;
  data: {
    total: number;
    list: ActivityLog[];
  };
}

export interface ActivityLogStatistics {
  total: number;
  operation_types: Array<{ _id: string; count: number }>;
  modules: Array<{ _id: string; count: number }>;
}

/**
 * 获取活动记录列表
 */
export async function getActivityLogs(params: ActivityLogQuery): Promise<ActivityLogResponse> {
  return request('/api/v1/activity-logs', {
    method: 'GET',
    params,
  });
}

/**
 * 获取最近的活动记录（用于仪表盘）
 */
export async function getRecentActivityLogs(limit: number = 5): Promise<ActivityLog[]> {
  return request('/api/v1/activity-logs/recent', {
    method: 'GET',
    params: { limit },
  });
}

/**
 * 获取活动记录详情
 */
export async function getActivityLogById(id: string): Promise<ActivityLog> {
  return request(`/api/v1/activity-logs/${id}`, {
    method: 'GET',
  });
}

/**
 * 获取活动记录统计
 */
export async function getActivityLogStatistics(days: number = 7): Promise<ActivityLogStatistics> {
  return request('/api/v1/activity-logs/statistics', {
    method: 'GET',
    params: { days },
  });
}

/**
 * 删除指定公司的活动记录（仅平台管理员）
 */
export async function deleteActivityLogsByCompany(companyId: string): Promise<void> {
  return request(`/api/v1/activity-logs/company/${companyId}`, {
    method: 'DELETE',
  });
}

// 操作类型常量
export const OPERATION_TYPES = {
  CREATE: 'create',
  UPDATE: 'update',
  DELETE: 'delete',
  VIEW: 'view',
  EXPORT: 'export',
  IMPORT: 'import',
  LOGIN: 'login',
  LOGOUT: 'logout',
} as const;

// 模块名称常量
export const MODULE_NAMES = {
  USER: '用户管理',
  ROLE: '角色管理',
  MENU: '菜单管理',
  COMPANY: '公司管理',
  POLICY: '保单管理',
  CUSTOMER: '客户管理',
  SYSTEM: '系统管理',
  AUTH: '认证授权',
} as const;

// 操作类型显示文本
export const OPERATION_TYPE_LABELS = {
  [OPERATION_TYPES.CREATE]: '新增',
  [OPERATION_TYPES.UPDATE]: '更新',
  [OPERATION_TYPES.DELETE]: '删除',
  [OPERATION_TYPES.VIEW]: '查看',
  [OPERATION_TYPES.EXPORT]: '导出',
  [OPERATION_TYPES.IMPORT]: '导入',
  [OPERATION_TYPES.LOGIN]: '登录',
  [OPERATION_TYPES.LOGOUT]: '登出',
} as const;

// 结果状态显示文本
export const RESULT_STATUS_LABELS = {
  success: '成功',
  failure: '失败',
} as const; 

File: internal/model/activity_log.go
package model

import (
	"time"

	"go.mongodb.org/mongo-driver/bson/primitive"
)

// ActivityLog 系统活动记录
type ActivityLog struct {
	ID            primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	UserID        string             `bson:"user_id" json:"user_id"`
	Username      string             `bson:"username" json:"username"`
	CompanyID     string             `bson:"company_id" json:"company_id"`
	CompanyName   string             `bson:"company_name" json:"company_name"`
	OperationType string             `bson:"operation_type" json:"operation_type"` // create/update/delete/view/export
	ModuleName    string             `bson:"module_name" json:"module_name"`
	OperationDesc string             `bson:"operation_desc" json:"operation_desc"`
	RequestURL    string             `bson:"request_url" json:"request_url"`
	RequestMethod string             `bson:"request_method" json:"request_method"`
	RequestParams interface{}        `bson:"request_params" json:"request_params"`
	IPAddress     string             `bson:"ip_address" json:"ip_address"`
	UserAgent     string             `bson:"user_agent" json:"user_agent"`
	OperationTime time.Time          `bson:"operation_time" json:"operation_time"`
	ExecutionTime int64              `bson:"execution_time" json:"execution_time"`     // 执行耗时(ms)
	ResultStatus  string             `bson:"result_status" json:"result_status"`       // success/failure
	TargetID      string             `bson:"target_id" json:"target_id,omitempty"`     // 操作目标ID
	TargetName    string             `bson:"target_name" json:"target_name,omitempty"` // 操作目标名称
}

// ActivityLogQuery 活动记录查询参数
type ActivityLogQuery struct {
	Page          int    `json:"page" form:"page"`
	PageSize      int    `json:"page_size" form:"page_size"`
	CompanyID     string `json:"company_id" form:"company_id"`
	UserID        string `json:"user_id" form:"user_id"`
	OperationType string `json:"operation_type" form:"operation_type"`
	ModuleName    string `json:"module_name" form:"module_name"`
	StartTime     string `json:"start_time" form:"start_time"`
	EndTime       string `json:"end_time" form:"end_time"`
}

// ActivityLogResponse 活动记录响应
type ActivityLogResponse struct {
	Total int64         `json:"total"`
	List  []ActivityLog `json:"list"`
}

// 操作类型常量
const (
	OperationTypeCreate = "create"
	OperationTypeUpdate = "update"
	OperationTypeDelete = "delete"
	OperationTypeView   = "view"
	OperationTypeExport = "export"
	OperationTypeImport = "import"
	OperationTypeLogin  = "login"
	OperationTypeLogout = "logout"
)

// 模块名称常量
const (
	ModuleUser     = "用户管理"
	ModuleRole     = "角色管理"
	ModuleMenu     = "菜单管理"
	ModuleCompany  = "公司管理"
	ModulePolicy   = "保单管理"
	ModuleCustomer = "客户管理"
	ModuleSystem   = "系统管理"
	ModuleAuth     = "认证授权"
)
